connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -101,6 +101,40 @@ class SparkConnectPlanner(val sessionHolder: SessionHolder) extends Logging {
   }
 
   // The root of the query plan is a relation and we apply the transformations to it.
+  def transformRelationAsDataset(rel: proto.Relation): Dataset[Row] = {
```

Comment:
This doesn't look making the code a lot better ... it basically saves the call of `Dataset.ofRows` in the individual transformation, which doesn't seem a lot of worthwhile.